db/migrate/20181001131632_add_conversion_host_id_to_miq_request_tasks.rb

```diff
@@ -0,0 +1,44 @@
+class AddConversionHostIdToMiqRequestTasks < ActiveRecord::Migration[5.0]
+  class MiqRequestTask < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    serialize :options, Hash
+    belongs_to :conversion_host, :class_name => "AddConversionHostIdToMiqRequestTasks::ConversionHost"
+  end
+
+  class ConversionHost < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class Host < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    add_column :miq_request_tasks, :conversion_host_id, :bigint
+
+    MiqRequestTask.where(:type => 'ServiceTemplateTransformationPlanTask').each do |task|
+      host_id = task.options[:transformation_host_id]
+      next unless host_id
+      host = Host.find_by(:id => host_id)
+      next if host.nil?
+      task.conversion_host = ConversionHost.find_or_create_by!(:resource_id => host.id, :resource_type => host.type) do |ch|
+        ch.name                     = host.name
+        ch.vddk_transport_supported = true
+        ch.ssh_transport_supported  = false
+      end
+      task.options.delete(:transformation_host_id)
+      task.save!
+    end
+  end
+
+  def down
+    MiqRequestTask.where(:type => 'ServiceTemplateTransformationPlanTask').each do |task|
+      next unless task.conversion_host
+      task.options[:transformation_host_id] = task.conversion_host.resource_id
+      ConversionHost.find(task.conversion_host.id).destroy!
```

Comment:
You can just pass the id to destroy like `ConversionHost.destroy(task.conversion_host.id)`

For performance it might be better to `.map` the conversion host ids that you want to destroy and issue one query after the loop like `conversion_host_ids = MiqRequestTask.where().map { |task| }; ConversionHost.destroy(conversion_host_ids)`

Reply:
Done.